lib/openhab/core/configuration.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require "forwardable"
+require_relative "emulate_hash"
+
+module OpenHAB
+  module Core
+    java_import org.openhab.core.config.core.Configuration
+
+    #
+    # {Configuration} represents openHAB's {org.openhab.core.config.core.Configuration} data
+    # with the full interface of {::Hash}.
+    #
+    # All keys are converted to strings.
+    #
+    class Configuration
+      include EmulateHash
+      extend Forwardable
+
+      def_delegator :get_properties, :to_hash
+      def_delegator :to_h, :inspect
+
+      # @!visibility private
+      alias_method :to_s, :inspect
+
+      # @!visibility private
+      def fetch(key, *default_value, &block)
+        if (result = get(key.to_s))
+          result
+        elsif block
+          yield key
+        elsif default_value.length > 1
+          raise ArgumentError, "Wrong number of arguments (given #{default_value.length + 1}, expected 1..2)"
+        elsif !default_value.empty?
+          default_value.first
+        else
+          raise KeyError, "Key not found #{key.inspect}"
+        end
+      end
+
+      # @!visibility private
+      def store(key, value)
+        put(key.to_s, value)
+      end
+
+      # @!visibility private
+      def delete(key)
```

Comment:
this does not implement the block param of https://docs.ruby-lang.org/en/3.1/Hash.html#method-i-delete

Reply:
Thanks, fixed now, but query: how thoroughly should we write our spec? Should it check for all these permutations?